flit_core/flit_core/buildapi.py

```diff
@@ -48,36 +45,36 @@ def prepare_metadata_for_build_wheel(metadata_directory, config_settings=None):
     module = Module(ini_info.module, Path.cwd())
     metadata = make_metadata(module, ini_info)
 
-    dist_info = osp.join(metadata_directory,
-                         dist_info_name(metadata.name, metadata.version))
-    os.mkdir(dist_info)
+    dist_info = Path(metadata_directory).joinpath(
+        dist_info_name(metadata.name, metadata.version))
+    dist_info.mkdir()
 
-    with io.open(osp.join(dist_info, 'WHEEL'), 'w', encoding='utf-8') as f:
+    with dist_info.joinpath('WHEEL').open('w', encoding='utf-8') as f:
         _write_wheel_file(f, supports_py2=metadata.supports_py2)
 
-    with io.open(osp.join(dist_info, 'METADATA'), 'w', encoding='utf-8') as f:
+    with dist_info.joinpath('METADATA').open('w', encoding='utf-8') as f:
         metadata.write_metadata_file(f)
 
     if ini_info.entrypoints:
-        with io.open(osp.join(dist_info, 'entry_points.txt'), 'w', encoding='utf-8') as f:
+        with dist_info.joinpath('entry_points.txt').open('w', encoding='utf-8') as f:
             write_entry_points(ini_info.entrypoints, f)
 
-    return osp.basename(dist_info)
+    return Path(dist_info).name
 
 # Metadata for editable are the same as for a wheel
 prepare_metadata_for_build_editable = prepare_metadata_for_build_wheel
 
 def build_wheel(wheel_directory, config_settings=None, metadata_directory=None):
     """Builds a wheel, places it in wheel_directory"""
-    info = make_wheel_in(pyproj_toml, Path(wheel_directory))
+    info = make_wheel_in(pyproj_toml, wheel_directory)
     return info.file.name
 
 def build_editable(wheel_directory, config_settings=None, metadata_directory=None):
     """Builds an "editable" wheel, places it in wheel_directory"""
-    info = make_wheel_in(pyproj_toml, Path(wheel_directory), editable=True)
+    info = make_wheel_in(pyproj_toml, wheel_directory, editable=True)
     return info.file.name
 
 def build_sdist(sdist_directory, config_settings=None):
     """Builds an sdist, places it in sdist_directory"""
-    path = SdistBuilder.from_ini_path(pyproj_toml).build(Path(sdist_directory))
+    path = SdistBuilder.from_ini_path(pyproj_toml).build(sdist_directory)
```

Comment:
I'd much rather that the various internals use `Path` objects, which are created here. That's much easier to reason about than thinking about the `str` objects coming in via this API.

Reply:
Fixed, forgot to revert that from a test I was doing.